var outer_caught = false;
var inner_caught = false;
try {
    try {
        raise("inner error");
    } catch e {
        inner_caught = true;
        assert(e == "inner error");
        raise("outer error");
    }
} catch e {
    outer_caught = true;
    assert(e == "outer error");
}
assert(inner_caught == true);
assert(outer_caught == true);